### ISSUE TITLE
Unicode.utf8Validate issue ?

### ISSUE THREAD

**Issue by passenger94** [gtkd-developers/GtkD#351]
Hi,
Many thanks for your work !!!

just used the utf8Validate function in glib/Unicode.d module
https://github.com/gtkd-developers/GtkD/blob/3ae42827345596b3d9ec6636d831418b2ed77b77/generated/gtkd/glib/Unicode.d#L1468

In my understanding `end` should be made of the chars from start of `str` to `outend`, `outend` beeing the first invalid char in `str` or pointing after the last char of `str` if valid.
At the moment the function is "returning" the chars from `outend` to the end of `str` or nothing.

```D

public static bool utf8Validate(string str, out string end)
{
    char* start = Str.toStringz(str), outend = null;

    auto __p = g_utf8_validate(start, cast(ptrdiff_t)str.length, &outend) != 0;

    end = Str.toString(start, outend - start);

    return __p;
}
	
unittest
{
    string valid;
    bool success = Unicode.utf8Validate("random text", valid);
    assert(success);
    assert(valid == "random text");
    success = Unicode.utf8Validate("random \xc3\x28text", valid);
    assert(!success);
    assert(valid == "random ");

    success = Unicode.utf8ValidateLen("random text", valid);
    assert(success);
    assert(valid == "random text");
    success = Unicode.utf8ValidateLen("random \xc3\x28text", valid);
    assert(!success);
    assert(valid == "random ");
}
```

	
Same goes for utf8ValidateLen() as seen in the unittest.
	
Did i miss something ?
Cheers

**Comment by passenger94** [comment]
Many Thanks !!

**Comment by dejlek** [comment]
@EsoCoding , why did you not simply ask to join the `gtkd-developers` group? Maybe there is no need to fork the project...

**Comment by EsoCoding** [comment]
Have been trying to get in contact, but even posting a simple issue wont be answered. The repo here has not been updated in a long time. Which is the case with many other d library's ans programs. I really like D but gtk is requirement for me and my projects, but without active development i may get trouble in the future. So thats why i took the matter in my own hands and now working on making gtk-d ready for gnome-builder. So it may get more interest from the new and interested developers. In the end its just a fork and if i am ready, i will surely try commit my work. Look at the really great work konstantIMP has done, his work should already have been merged a long time ago. He has been faf ahead of gtkd, but hes work is hard to find. So i am promoting he's work now preparing to add D language and gtkd into gnome builder. I think D is a underestimated language and gtkd can be a door to this language.

**Comment by hrauchgh** [comment]
Where has your fork gone? I just wanted to look at your new gtk-d repo, but I get a 404 when I use the link you mentioned above.

**Comment by sltu** [comment]
Yes, it's true that D is an underestimated language, and gtkd can be a gateway to it. However, if there are no good and well-maintained GTK bindings, the purpose of using the D language diminishes.

And now, we have two options: the first one is unmaintained (this repository), and the second one returns a 404 error.
